src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataImportService.cs

```diff
@@ -74,38 +76,42 @@ public async Task<bool> HasIncompleteImports(Guid releaseId)
                 .AnyAsync(import => import.Status != DataImportStatus.COMPLETE);
         }
 
-        public async Task<DataImportViewModel> GetImport(Guid fileId)
+        public async Task<DataImportStatusViewModel> GetImportStatus(Guid fileId)
         {
             var import = await _dataImportRepository.GetByFileId(fileId);
 
             if (import == null)
             {
-                return DataImportViewModel.NotFound();
+                return DataImportStatusViewModel.NotFound();
             }
 
-            return new DataImportViewModel
+            await _contentDbContext.Entry(import)
+                .Collection(i => i.Errors)
+                .LoadAsync();
+
+            return new DataImportStatusViewModel
             {
                 Errors = import.Errors.Select(error => error.Message).ToList(),
                 PercentageComplete = import.PercentageComplete(),
                 StagePercentageComplete = import.StagePercentageComplete,
-                NumberOfRows = import.Rows,
+                NumberOfRows = import.Rows > 0 ? import.Rows : null,
```

Comment:
No never negative but I wanted to avoid returning a value of 0 as that's not strictly true, the importer just hasn't processed the data or zip file yet so we don't know the number of rows.

The value isn't set when the `DataImport` is created by the admin but the field isn't nullable so it's initialised as 0.

To display the number of rows as 'Unknown` rather than 0 and avoid the expression you've highlighted here I've made `TotalRows` nullable along with a few other changes to `DataImport`. I didn't change everything as I wanted to keep Importer changes to a minimum.

Changes are on [this commit](https://github.com/dfe-analytical-services/explore-education-statistics/pull/3387/commits/80c29ce1c22cc19f607af57ff647d76d9bd7213c).